src/course/application/course-get-by-user/course-get-by-user.ts

```diff
@@ -0,0 +1,41 @@
+import { EnrollmentRepository } from '../../domain/repository/enrollment-repository';
+import { CourseRepository } from '../../domain/repository/course-repository';
+import { UserId } from '../../../user/domain/user-id';
+import { Course } from '../../domain/course';
+
+interface CourseWithEnrollment {
+  course: Course;
+  enrolledAt: Date;
+}
+
+export class CourseGetByUser {
+  constructor(
+    private enrollmentRepository: EnrollmentRepository,
+    private courseRepository: CourseRepository
+  ) {}
+
+  async execute(userId: string): Promise<CourseWithEnrollment[]> {
+    const enrollments = await this.enrollmentRepository.findByUser(
+      new UserId(userId)
+    );
+
+    const coursesWithEnrollment = await Promise.all(
+      enrollments.map(async enrollment => {
+        const course = await this.courseRepository.getById(enrollment.courseId);
+
+        if (!course) {
+          throw new Error(
```

Comment:
When a course is missing the code throws a generic Error. Throwing generic errors makes it harder to differentiate domain errors and handle them appropriately at controller level.

**Sugerencia:**
```typescript
Throw a specific domain error (e.g., CourseNotFoundError) so controllers can map it to 404 consistently. Example:
if (!course) throw new CourseNotFoundError(`Course with id ${enrollment.courseId.getValue()} not found`);
```